torch/lib/c10d/logger.cpp

```diff
@@ -85,6 +87,101 @@ void Logger::set_construction_logging_data(
   LogPyTorchDDPUsage(*ddp_logging_data_);
 }
 
+void Logger::calculate_avg_cpu_time(
+    int64_t& avg_time,
+    int64_t cpu_start_time,
+    int64_t cpu_end_time) {
+  long num_iters = reducer_->num_iterations_;
+  avg_time = ((cpu_end_time - cpu_start_time) + avg_time * (num_iters - 1)) /
+      num_iters;
+}
+
+#ifdef USE_CUDA
+void Logger::calculate_avg_gpu_time(
+    int64_t& avg_time,
+    cudaEvent_t gpu_start,
+    cudaEvent_t gpu_end) {
+  long num_iters = reducer_->num_iterations_;
+  float milliseconds = 0.0;
+  cudaEventElapsedTime(&milliseconds, gpu_start, gpu_end);
+  avg_time =
+      (int(milliseconds * 1000000) + avg_time * (num_iters - 1)) / num_iters;
```

Comment:
Nit: Use constant instead of magic number

Reply:
sounds good